File: speedsolutionschallenge.client/src/services/dispenserService.js
const API_URL = 'api/dispenser';

const dispenserService = {
    createDispenser: async (dispenserData) => {
        try {
            const response = await fetch(`${API_URL}/create`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(dispenserData),
            });

            const data = await response.json();
            return data;
        } catch (error) {
            throw new Error(`Error al crear un dispensador: ${error.message}`);
        }
    },

    getAllDispensers: async () => {
        try {
            const response = await fetch(`${API_URL}/getAll`);
            const data = await response.json();
            return data;
        } catch (error) {
            throw new Error(`Error al obtener los dispensadores: ${error.message}`);
        }
    },

    getDispenserById: async (dispenserId) => {
        try {
            const response = await fetch(`${API_URL}/getById/${dispenserId}`);
            const data = await response.json();
            return data;
        } catch (error) {
            throw new Error(`Error al obtener el dispensador: ${error.message}`);
        }
    },

    updateDispenser: async (dispenserId, updatedDispenser) => {
        try {
            const response = await fetch(`${API_URL}/update/${dispenserId}`, {
                method: 'PUT',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(updatedDispenser),
            });

            const data = await response.json();
            return data;
        } catch (error) {
            throw new Error(`Error al actualizar el dispensador: ${error.message}`);
        }
    },
};

export default dispenserService;

File: SpeedSolutionsChallenge.Server/Services/Dispenser/DispenserService.cs
using SpeedSolutionsChallenge.Data.Models;
using SpeedSolutionsChallenge.Data.Repositories.DispenserRepository;

namespace SpeedSolutionsChallenge.Server.Services.DispenserService
{
    public class DispenserService : IDispenserService
    {
        private readonly IDispenserRepository _dispenserRepository;

        public DispenserService(IDispenserRepository dispenserRepository)
        {
            _dispenserRepository = dispenserRepository;
        }

        public async Task<Dispenser> CreateDispenser(Dispenser dispenser)
        {
            return await _dispenserRepository.CreateDispenser(dispenser);
        }

        public async Task<List<Dispenser>> GetAllDispensers()
        {
            return await _dispenserRepository.GetAllDispensers();
        }

        public async Task<Dispenser> GetDispenserById(int dispenserId)
        {
            return await _dispenserRepository.GetDispenserById(dispenserId);
        }

        public async Task<Dispenser> UpdateDispenser(int dispenserId, Dispenser updatedDispenser)
        {
            return await _dispenserRepository.UpdateDispenser(dispenserId, updatedDispenser);
        }
    }
}
